src/protocols/minecraft/types.rs

```diff
@@ -156,6 +156,20 @@ impl JavaResponse {
     }
 }
 
+pub struct RequestSettings {
+    pub hostname: String,
+    pub protocol_version: i32,
```

Comment:
It might be worth adding a doc comment about -1 meaning any version, like you mention in the mc example